lib/improver/weighted_blend.py

```diff
@@ -57,20 +252,31 @@ def __init__(self, coord, coord_adjust=None):
     def __repr__(self):
         """Represent the configured plugin instance as a string."""
         return (
-            '<BasicWeightedAverage: coord = {0:s}>').format(self.coord)
+            '<WeightedBlend: coord = {0:s}>').format(self.coord)
 
     def process(self, cube, weights=None):
-        """Calculate weighted mean across the chosen coord
+        """Calculate weighted blend across the chosen coord
 
         Args:
             cube : iris.cube.Cube
                    Cube to blend across the coord.
             weights: Optional list or np.array of weights
-                     or None (equivalent to equal weights)
+                     or None (equivalent to equal weights).
 
         Returns:
             result : iris.cube.Cube
-                     containing the weighted mean across the chosen coord
+                     containing the weighted blend across the chosen coord.
+
+        Raises:
+            ValueError 1 : first argument not a cube.
```

Comment:
Yes I think that's true. @benfitzpatrick  could you give some guidance?

I think value error 2 could be removed as iris will raise a sensible error on its own. I think this could be true of value error 1 too. This would mean there are slightly less errors we are raising.